[deconz] OH2.5.10 high CPU usage after sending ON/OFF commands to item.

## Expected Behavior
Not using 80-100% CPU for 10-30s after sending a single item a single command.
Not going offline and occasionally crash after sending a single item a short burst of commands.

## Current Behavior
Since the deconz HUE binding implementation is broken #9175 I started using the deconz binding instead.
After experiencing strange delays and also a total crash of the binding requiring a karaf restart of it, I started investigating.

Using this item:

```
Switch BulbStairsState            (livingRoom, lights)   { channel = "deconz:colortemperaturelight:homeserver:000b57fffe103e8101:brightness" }
Dimmer BulbStairsBright           (livingRoom)           { channel = "deconz:colortemperaturelight:homeserver:000b57fffe103e8101:brightness" }
Dimmer BulbStairsColorTemp        (livingRoom)           { channel = "deconz:colortemperaturelight:homeserver:000b57fffe103e8101:color_temperature" }
```

sending this command from a rule:

```
if (BulbStairsState.state == ON) { 
    BulbStairsState.sendCommand(OFF) 
} else { 
    BulbStairsState.sendCommand(ON) 
}
```

Causes the java process to consume 80-100% CPU.
(With PaperUI open displaying Configuration/Things Search dec the CPU usage increases to 150-190% CPU.)

In this case the system recovers after 10-30s.

I have experienced lately that deconz items do not work in the morning, and have traced that down to my evening rule shutting off everything. After that the deconz binding stops executing and the deconz bridge goes and stays offline.

In order the reproduce with a simple test, I did the following:

```
if (BulbStairsState.state == ON) { 
    BulbStairsState.sendCommand(OFF) 
    BulbStairsState.sendCommand(OFF) 
    BulbStairsState.sendCommand(OFF) 
    BulbStairsState.sendCommand(OFF) 
    BulbStairsState.sendCommand(OFF) 
} else { 
    BulbStairsState.sendCommand(ON) 
    BulbStairsState.sendCommand(ON) 
    BulbStairsState.sendCommand(ON) 
    BulbStairsState.sendCommand(ON) 
    BulbStairsState.sendCommand(ON) 
}
```

Now, the bridge goes offline almost immediately, sometimes recovering, sometimes not.
System takes a lot longer to recover if the binding recovers, but with PaperUI open, it never recovers until Chrome Tab is closed.

## Possible Solution

## Steps to Reproduce (for Bugs)
See above.

## Context

## Your Environment
<!-- Include as many relevant details about the environment you experienced the bug in -->
* Version used: openhab 2.5.10 with matching addons. Deconz headless 2.5.86 (same on 2.6.0)
* Environment: java --version
openjdk 11.0.9.1 2020-11-04
OpenJDK Runtime Environment (build 11.0.9.1+1-Ubuntu-0ubuntu1.18.04)
OpenJDK 64-Bit Server VM (build 11.0.9.1+1-Ubuntu-0ubuntu1.18.04, mixed mode, sharing)
* Ubuntu 18.04 LTS x86_64 (2 cores)

Deconz debug log when sending short burst that causes the binding to go offline:

```
2020-12-06 16:50:12.853 [DEBUG] [nternal.netutils.WebSocketConnection] - Connect: /127.0.0.1
2020-12-06 16:50:12.859 [DEBUG] [nternal.netutils.WebSocketConnection] - Connect: /127.0.0.1
2020-12-06 16:50:12.862 [DEBUG] [nternal.netutils.WebSocketConnection] - Connect: /127.0.0.1
2020-12-06 16:50:12.865 [DEBUG] [nternal.netutils.WebSocketConnection] - Connect: /127.0.0.1
2020-12-06 16:50:12.870 [DEBUG] [nternal.netutils.WebSocketConnection] - Connect: /127.0.0.1
2020-12-06 16:50:12.867 [DEBUG] [nternal.netutils.WebSocketConnection] - Connect: /127.0.0.1
2020-12-06 16:50:12.866 [DEBUG] [nternal.netutils.WebSocketConnection] - Connect: /127.0.0.1
2020-12-06 16:50:12.889 [DEBUG] [z.internal.handler.LightThingHandler] - val = '201', scaledValue = '80'
2020-12-06 16:50:12.872 [DEBUG] [nternal.netutils.WebSocketConnection] - Connect: /127.0.0.1
2020-12-06 16:50:13.011 [INFO ] [clipse.smarthome.model.script.Stairs] - BulbStairsBright.state=80
2020-12-06 16:50:14.890 [DEBUG] [ernal.handler.DeconzBaseThingHandler] - Get new state failed:
java.util.concurrent.CompletionException: java.util.concurrent.TimeoutException: Total timeout 2000 ms elapsed
        at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:331) ~[?:?]
        at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:346) ~[?:?]
        at java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:632) ~[?:?]
        at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506) [?:?]
        at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2088) [?:?]
        at org.openhab.binding.deconz.internal.netutils.AsyncHttpClient$1.onComplete(AsyncHttpClient.java:109) [bundleFile:?]
        at org.eclipse.jetty.client.ResponseNotifier.notifyComplete(ResponseNotifier.java:198) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.client.ResponseNotifier.notifyComplete(ResponseNotifier.java:190) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.client.HttpReceiver.terminateResponse(HttpReceiver.java:444) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.client.HttpReceiver.abort(HttpReceiver.java:527) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.client.HttpChannel.abortResponse(HttpChannel.java:156) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.client.HttpChannel.abort(HttpChannel.java:149) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.client.HttpExchange.abort(HttpExchange.java:257) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.client.HttpConversation.abort(HttpConversation.java:149) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.client.HttpRequest.abort(HttpRequest.java:768) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.client.TimeoutCompleteListener.onTimeoutExpired(TimeoutCompleteListener.java:51) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.client.TimeoutCompleteListener.schedule(TimeoutCompleteListener.java:73) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.client.HttpChannel.send(HttpChannel.java:126) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.client.HttpConnection.send(HttpConnection.java:201) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.client.http.HttpConnectionOverHTTP$Delegate.send(HttpConnectionOverHTTP.java:252) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.client.http.HttpConnectionOverHTTP.send(HttpConnectionOverHTTP.java:121) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.client.http.HttpDestinationOverHTTP.send(HttpDestinationOverHTTP.java:38) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.client.HttpDestination.process(HttpDestination.java:346) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.client.HttpDestination.process(HttpDestination.java:304) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.client.HttpDestination.send(HttpDestination.java:294) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.client.HttpDestination.succeeded(HttpDestination.java:228) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.client.AbstractConnectionPool.proceed(AbstractConnectionPool.java:153) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.client.AbstractConnectionPool$1.succeeded(AbstractConnectionPool.java:131) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.client.AbstractConnectionPool$1.succeeded(AbstractConnectionPool.java:123) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.util.Promise$Wrapper.succeeded(Promise.java:130) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.client.http.HttpConnectionOverHTTP.onOpen(HttpConnectionOverHTTP.java:129) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.io.SelectorManager.connectionOpened(SelectorManager.java:324) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.io.ManagedSelector.createEndPoint(ManagedSelector.java:270) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.io.ManagedSelector.access$1900(ManagedSelector.java:61) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.io.ManagedSelector$CreateEndPoint.run(ManagedSelector.java:923) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:782) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:918) [bundleFile:9.4.20.v20190813]
        at java.lang.Thread.run(Thread.java:834) [?:?]
Caused by: java.util.concurrent.TimeoutException: Total timeout 2000 ms elapsed
        ... 23 more
2020-12-06 16:50:14.888 [DEBUG] [ernal.handler.DeconzBaseThingHandler] - Get new state failed:
java.util.concurrent.CompletionException: java.util.concurrent.TimeoutException: Total timeout 2000 ms elapsed
        at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:331) ~[?:?]
        at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:346) ~[?:?]
        at java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:632) ~[?:?]
        at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506) [?:?]

```

Here is the first part:

```
2020-12-06 22:09:57.722 [INFO ] [e.smarthome.model.script.KeyFob1Rule] - Update: 2.0
2020-12-06 22:09:57.723 [INFO ] [e.smarthome.model.script.KeyFob1Rule] - O1xClick
2020-12-06 22:09:57.750 [TRACE] [z.internal.handler.LightThingHandler] - Sending {"on":false} to light 48 via http://127.0.0.1:80/api/0950775C33/lights/48/state
2020-12-06 22:09:57.751 [TRACE] [z.internal.handler.LightThingHandler] - Result code=200, body=[{"success":{"/lights/48/state/on":false}}]
2020-12-06 22:09:57.753 [TRACE] [nternal.netutils.WebSocketConnection] - Raw data received by websocket: {"e":"changed","id":"48","r":"lights","state":{"alert":null,"bri":201,"colormode":"ct","ct":398,"on":false,"reachable":true},"t":"event","uniqueid":"00:0b:57:ff:fe:10:3e:81-01"}
2020-12-06 22:09:57.753 [TRACE] [nternal.netutils.WebSocketConnection] - Raw data received by websocket: {"e":"changed","id":"48","r":"lights","state":{"alert":null,"bri":201,"colormode":"ct","ct":398,"on":false,"reachable":true},"t":"event","uniqueid":"00:0b:57:ff:fe:10:3e:81-01"}
2020-12-06 22:09:57.753 [TRACE] [z.internal.handler.LightThingHandler] - deconz:colortemperaturelight:homeserver:000b57fffe103e8101 received LightMessage{hascolor=null, ctmax=null, ctmin=null, type=null, state=LightState{reachable=true, on=false, bri=201, alert='null', colormode='ct', effect='null', hue=null, sat=null, ct=398, xy=null, transitiontime=null}, e='changed', r='lights', t='event', id='48', manufacturername='', modelid='', name='', swversion='', ep='', uniqueid='00:0b:57:ff:fe:10:3e:81-01'}
2020-12-06 22:09:57.754 [TRACE] [z.internal.handler.LightThingHandler] - deconz:colortemperaturelight:homeserver:000b57fffe103e8101 received LightMessage{hascolor=null, ctmax=null, ctmin=null, type=null, state=LightState{reachable=true, on=false, bri=201, alert='null', colormode='ct', effect='null', hue=null, sat=null, ct=398, xy=null, transitiontime=null}, e='changed', r='lights', t='event', id='48', manufacturername='', modelid='', name='', swversion='', ep='', uniqueid='00:0b:57:ff:fe:10:3e:81-01'}
2020-12-06 22:09:57.754 [TRACE] [nternal.netutils.WebSocketConnection] - Raw data received by websocket: {"e":"changed","id":"48","r":"lights","state":{"alert":null,"bri":201,"colormode":"ct","ct":398,"on":false,"reachable":true},"t":"event","uniqueid":"00:0b:57:ff:fe:10:3e:81-01"}
2020-12-06 22:09:57.754 [TRACE] [z.internal.handler.LightThingHandler] - deconz:colortemperaturelight:homeserver:000b57fffe103e8101 received LightMessage{hascolor=null, ctmax=null, ctmin=null, type=null, state=LightState{reachable=true, on=false, bri=201, alert='null', colormode='ct', effect='null', hue=null, sat=null, ct=398, xy=null, transitiontime=null}, e='changed', r='lights', t='event', id='48', manufacturername='', modelid='', name='', swversion='', ep='', uniqueid='00:0b:57:ff:fe:10:3e:81-01'}
2020-12-06 22:09:57.754 [TRACE] [nternal.netutils.WebSocketConnection] - Raw data received by websocket: {"e":"changed","id":"48","r":"lights","state":{"alert":null,"bri":201,"colormode":"ct","ct":398,"on":false,"reachable":true},"t":"event","uniqueid":"00:0b:57:ff:fe:10:3e:81-01"}
2020-12-06 22:09:57.754 [TRACE] [nternal.netutils.WebSocketConnection] - Raw data received by websocket: {"e":"changed","id":"48","r":"lights","state":{"alert":null,"bri":201,"colormode":"ct","ct":398,"on":false,"reachable":true},"t":"event","uniqueid":"00:0b:57:ff:fe:10:3e:81-01"}
2020-12-06 22:09:57.756 [INFO ] [clipse.smarthome.model.script.Stairs] - BulbStairsBright.state=0
2020-12-06 22:09:57.755 [TRACE] [z.internal.handler.LightThingHandler] - deconz:colortemperaturelight:homeserver:000b57fffe103e8101 received LightMessage{hascolor=null, ctmax=null, ctmin=null, type=null, state=LightState{reachable=true, on=false, bri=201, alert='null', colormode='ct', effect='null', hue=null, sat=null, ct=398, xy=null, transitiontime=null}, e='changed', r='lights', t='event', id='48', manufacturername='', modelid='', name='', swversion='', ep='', uniqueid='00:0b:57:ff:fe:10:3e:81-01'}
2020-12-06 22:09:57.757 [TRACE] [z.internal.handler.LightThingHandler] - deconz:colortemperaturelight:homeserver:000b57fffe103e8101 received LightMessage{hascolor=null, ctmax=null, ctmin=null, type=null, state=LightState{reachable=true, on=false, bri=201, alert='null', colormode='ct', effect='null', hue=null, sat=null, ct=398, xy=null, transitiontime=null}, e='changed', r='lights', t='event', id='48', manufacturername='', modelid='', name='', swversion='', ep='', uniqueid='00:0b:57:ff:fe:10:3e:81-01'}
2020-12-06 22:09:57.762 [DEBUG] [nternal.netutils.WebSocketConnection] - Connecting to: ws://127.0.0.1:443
2020-12-06 22:09:57.763 [DEBUG] [nternal.netutils.WebSocketConnection] - Connecting to: ws://127.0.0.1:443
2020-12-06 22:09:57.763 [DEBUG] [nternal.netutils.WebSocketConnection] - Connecting to: ws://127.0.0.1:443
2020-12-06 22:09:57.763 [DEBUG] [nternal.netutils.WebSocketConnection] - Connecting to: ws://127.0.0.1:443
2020-12-06 22:09:57.765 [DEBUG] [nternal.netutils.WebSocketConnection] - Connecting to: ws://127.0.0.1:443
2020-12-06 22:09:57.769 [DEBUG] [nternal.netutils.WebSocketConnection] - Connect: /127.0.0.1
2020-12-06 22:09:57.769 [DEBUG] [nternal.netutils.WebSocketConnection] - Connect: /127.0.0.1
2020-12-06 22:09:57.770 [DEBUG] [nternal.netutils.WebSocketConnection] - Connect: /127.0.0.1
2020-12-06 22:09:57.776 [TRACE] [ernal.handler.DeconzBaseThingHandler] - Requesting URL for initial data: http://127.0.0.1:80/api/0950775C33/lights/79
2020-12-06 22:09:57.776 [TRACE] [ernal.handler.DeconzBaseThingHandler] - Requesting URL for initial data: http://127.0.0.1:80/api/0950775C33/lights/38
2020-12-06 22:09:57.779 [TRACE] [ernal.handler.DeconzBaseThingHandler] - Requesting URL for initial data: http://127.0.0.1:80/api/0950775C33/lights/48
2020-12-06 22:09:57.770 [DEBUG] [nternal.netutils.WebSocketConnection] - Connect: /127.0.0.1
2020-12-06 22:09:57.780 [TRACE] [z.internal.handler.LightThingHandler] - deconz:colortemperaturelight:homeserver:000b57fffe103e8101 received LightMessage{hascolor=true, ctmax=454, ctmin=250, type=COLOR_TEMPERATURE_LIGHT, state=LightState{reachable=true, on=false, bri=201, alert='none', colormode='ct', effect='null', hue=null, sat=null, ct=398, xy=null, transitiontime=null}, e='', r='', t='', id='', manufacturername='IKEA of Sweden', modelid='TRADFRI bulb E27 WS opal 980lm', name='Stue pendel', swversion='1.2.217', ep='', uniqueid='00:0b:57:ff:fe:10:3e:81-01'}
2020-12-06 22:09:57.782 [TRACE] [ernal.handler.DeconzBaseThingHandler] - Requesting URL for initial data: http://127.0.0.1:80/api/0950775C33/sensors/28
2020-12-06 22:09:57.772 [DEBUG] [nternal.netutils.WebSocketConnection] - Connect: /127.0.0.1
2020-12-06 22:09:57.778 [TRACE] [z.internal.handler.LightThingHandler] - deconz:colortemperaturelight:homeserver:0017880108440a1f0b received LightMessage{hascolor=true, ctmax=65279, ctmin=0, type=COLOR_TEMPERATURE_LIGHT, state=LightState{reachable=true, on=false, bri=5, alert='none', colormode='ct', effect='null', hue=null, sat=null, ct=454, xy=null, transitiontime=null}, e='', r='', t='', id='', manufacturername='Philips', modelid='LTG002', name='MDY bed 79', swversion='1.55.7_r28193', ep='', uniqueid='00:17:88:01:08:44:0a:1f-0b'}
2020-12-06 22:09:57.784 [TRACE] [z.internal.handler.LightThingHandler] - deconz:onofflight:homeserver:d0cf5efffe7ab6b801 received LightMessage{hascolor=false, ctmax=null, ctmin=null, type=ON_OFF_PLUGIN_UNIT, state=LightState{reachable=true, on=true, bri=null, alert='none', colormode='null', effect='null', hue=null, sat=null, ct=null, xy=null, transitiontime=null}, e='', r='', t='', id='', manufacturername='IKEA of Sweden', modelid='TRADFRI control outlet', name='Boom arbeidsbenk/38', swversion='1.4.020', ep='', uniqueid='d0:cf:5e:ff:fe:7a:b6:b8-01'}
2020-12-06 22:09:57.776 [TRACE] [ernal.handler.DeconzBaseThingHandler] - Requesting URL for initial data: http://127.0.0.1:80/api/0950775C33/lights/78
2020-12-06 22:09:57.775 [TRACE] [ernal.handler.DeconzBaseThingHandler] - Requesting URL for initial data: http://127.0.0.1:80/api/0950775C33/lights/70
2020-12-06 22:09:57.785 [TRACE] [ernal.handler.DeconzBaseThingHandler] - Requesting URL for initial data: http://127.0.0.1:80/api/0950775C33/lights/68
2020-12-06 22:09:57.786 [TRACE] [ernal.handler.DeconzBaseThingHandler] - Requesting URL for initial data: http://127.0.0.1:80/api/0950775C33/sensors/25
2020-12-06 22:09:57.786 [TRACE] [ernal.handler.DeconzBaseThingHandler] - Requesting URL for initial data: http://127.0.0.1:80/api/0950775C33/lights/69
2020-12-06 22:09:57.786 [TRACE] [ernal.handler.DeconzBaseThingHandler] - Requesting URL for initial data: http://127.0.0.1:80/api/0950775C33/sensors/34
2020-12-06 22:09:57.786 [TRACE] [ernal.handler.DeconzBaseThingHandler] - Requesting URL for initial data: http://127.0.0.1:80/api/0950775C33/sensors/26
2020-12-06 22:09:57.787 [TRACE] [ernal.handler.DeconzBaseThingHandler] - Requesting URL for initial data: http://127.0.0.1:80/api/0950775C33/sensors/12
2020-12-06 22:09:57.787 [TRACE] [ernal.handler.DeconzBaseThingHandler] - Requesting URL for initial data: http://127.0.0.1:80/api/0950775C33/sensors/16
2020-12-06 22:09:57.787 [TRACE] [ernal.handler.DeconzBaseThingHandler] - Requesting URL for initial data: http://127.0.0.1:80/api/0950775C33/sensors/11
2020-12-06 22:09:57.787 [TRACE] [ernal.handler.DeconzBaseThingHandler] - Requesting URL for initial data: http://127.0.0.1:80/api/0950775C33/sensors/15
2020-12-06 22:09:57.787 [TRACE] [ernal.handler.DeconzBaseThingHandler] - Requesting URL for initial data: http://127.0.0.1:80/api/0950775C33/sensors/33
2020-12-06 22:09:57.787 [TRACE] [ernal.handler.DeconzBaseThingHandler] - Requesting URL for initial data: http://127.0.0.1:80/api/0950775C33/sensors/1
2020-12-06 22:09:57.788 [TRACE] [ernal.handler.DeconzBaseThingHandler] - Requesting URL for initial data: http://127.0.0.1:80/api/0950775C33/sensors/5
2020-12-06 22:09:57.788 [TRACE] [ernal.handler.DeconzBaseThingHandler] - Requesting URL for initial data: http://127.0.0.1:80/api/0950775C33/sensors/17
2020-12-06 22:09:57.788 [TRACE] [ernal.handler.DeconzBaseThingHandler] - Requesting URL for initial data: http://127.0.0.1:80/api/0950775C33/sensors/3
2020-12-06 22:09:57.788 [TRACE] [ernal.handler.DeconzBaseThingHandler] - Requesting URL for initial data: http://127.0.0.1:80/api/0950775C33/sensors/2
2020-12-06 22:09:57.788 [TRACE] [ernal.handler.DeconzBaseThingHandler] - Requesting URL for initial data: http://127.0.0.1:80/api/0950775C33/sensors/24
2020-12-06 22:09:57.771 [TRACE] [ernal.handler.DeconzBaseThingHandler] - Requesting URL for initial data: http://127.0.0.1:80/api/0950775C33/sensors/30
2020-12-06 22:09:57.788 [TRACE] [z.internal.handler.LightThingHandler] - deconz:extendedcolorlight:homeserver:00178801049a5adb0b received LightMessage{hascolor=true, ctmax=500, ctmin=153, type=EXTENDED_COLOR_LIGHT, state=LightState{reachable=true, on=false, bri=152, alert='none', colormode='ct', effect='none', hue=8418, sat=140, ct=366, xy=[0.4573, 0.41], transitiontime=null}, e='', r='', t='', id='', manufacturername='Philips', modelid='LST002', name='HUE Strip Baderomsinnredning 68', swversion='1.24.0_r20239', ep='', uniqueid='00:17:88:01:04:9a:5a:db-0b'}
2020-12-06 22:09:57.788 [TRACE] [ernal.handler.DeconzBaseThingHandler] - Requesting URL for initial data: http://127.0.0.1:80/api/0950775C33/sensors/4
2020-12-06 22:09:57.789 [TRACE] [ernal.handler.DeconzBaseThingHandler] - Requesting URL for initial data: http://127.0.0.1:80/api/0950775C33/sensors/13
2020-12-06 22:09:57.789 [TRACE] [ernal.handler.DeconzBaseThingHandler] - Requesting URL for initial data: http://127.0.0.1:80/api/0950775C33/sensors/27
2020-12-06 22:09:57.789 [TRACE] [ernal.handler.DeconzBaseThingHandler] - Requesting URL for initial data: http://127.0.0.1:80/api/0950775C33/sensors/19
2020-12-06 22:09:57.789 [TRACE] [ernal.handler.DeconzBaseThingHandler] - Requesting URL for initial data: http://127.0.0.1:80/api/0950775C33/sensors/18
2020-12-06 22:09:57.790 [TRACE] [ernal.handler.DeconzBaseThingHandler] - Requesting URL for initial data: http://127.0.0.1:80/api/0950775C33/sensors/6
2020-12-06 22:09:57.789 [TRACE] [ernal.handler.DeconzBaseThingHandler] - Requesting URL for initial data: http://127.0.0.1:80/api/0950775C33/sensors/7
2020-12-06 22:09:57.777 [TRACE] [ernal.handler.DeconzBaseThingHandler] - Requesting URL for initial data: http://127.0.0.1:80/api/0950775C33/sensors/23
2020-12-06 22:09:57.791 [TRACE] [ernal.handler.DeconzBaseThingHandler] - Requesting URL for initial data: http://127.0.0.1:80/api/0950775C33/sensors/30
2020-12-06 22:09:57.791 [TRACE] [ernal.handler.DeconzBaseThingHandler] - Requesting URL for initial data: http://127.0.0.1:80/api/0950775C33/lights/79
2020-12-06 22:09:57.791 [TRACE] [ernal.handler.DeconzBaseThingHandler] - Requesting URL for initial data: http://127.0.0.1:80/api/0950775C33/lights/38
2020-12-06 22:09:57.791 [TRACE] [ernal.handler.DeconzBaseThingHandler] - Requesting URL for initial data: http://127.0.0.1:80/api/0950775C33/lights/78
2020-12-06 22:09:57.792 [TRACE] [ernal.handler.DeconzBaseThingHandler] - Requesting URL for initial data: http://127.0.0.1:80/api/0950775C33/sensors/23
2020-12-06 22:09:57.792 [TRACE] [ernal.handler.DeconzBaseThingHandler] - Requesting URL for initial data: http://127.0.0.1:80/api/0950775C33/lights/48
2020-12-06 22:09:57.792 [TRACE] [ernal.handler.DeconzBaseThingHandler] - Requesting URL for initial data: http://127.0.0.1:80/api/0950775C33/sensors/28
2020-12-06 22:09:57.792 [TRACE] [ernal.handler.DeconzBaseThingHandler] - Requesting URL for initial data: http://127.0.0.1:80/api/0950775C33/lights/68
2020-12-06 22:09:57.792 [TRACE] [ernal.handler.DeconzBaseThingHandler] - Requesting URL for initial data: http://127.0.0.1:80/api/0950775C33/sensors/25
2020-12-06 22:09:57.792 [TRACE] [ernal.handler.DeconzBaseThingHandler] - Requesting URL for initial data: http://127.0.0.1:80/api/0950775C33/lights/69
2020-12-06 22:09:57.793 [TRACE] [ernal.handler.DeconzBaseThingHandler] - Requesting URL for initial data: http://127.0.0.1:80/api/0950775C33/sensors/34
2020-12-06 22:09:57.793 [TRACE] [ernal.handler.DeconzBaseThingHandler] - Requesting URL for initial data: http://127.0.0.1:80/api/0950775C33/sensors/26
2020-12-06 22:09:57.793 [TRACE] [ernal.handler.DeconzBaseThingHandler] - Requesting URL for initial data: http://127.0.0.1:80/api/0950775C33/sensors/12
2020-12-06 22:09:57.793 [TRACE] [ernal.handler.DeconzBaseThingHandler] - Requesting URL for initial data: http://127.0.0.1:80/api/0950775C33/sensors/16
2020-12-06 22:09:57.793 [TRACE] [ernal.handler.DeconzBaseThingHandler] - Requesting URL for initial data: http://127.0.0.1:80/api/0950775C33/sensors/11
2020-12-06 22:09:57.794 [TRACE] [ernal.handler.DeconzBaseThingHandler] - Requesting URL for initial data: http://127.0.0.1:80/api/0950775C33/sensors/15
2020-12-06 22:09:57.794 [TRACE] [ernal.handler.DeconzBaseThingHandler] - Requesting URL for initial data: http://127.0.0.1:80/api/0950775C33/sensors/33
2020-12-06 22:09:57.794 [TRACE] [ernal.handler.DeconzBaseThingHandler] - Requesting URL for initial data: http://127.0.0.1:80/api/0950775C33/sensors/1
2020-12-06 22:09:57.794 [TRACE] [ernal.handler.DeconzBaseThingHandler] - Requesting URL for initial data: http://127.0.0.1:80/api/0950775C33/sensors/5
2020-12-06 22:09:57.794 [TRACE] [ernal.handler.DeconzBaseThingHandler] - Requesting URL for initial data: http://127.0.0.1:80/api/0950775C33/sensors/17
2020-12-06 22:09:57.793 [TRACE] [z.internal.handler.LightThingHandler] - deconz:dimmablelight:homeserver:680ae2fffe3a34a501 received LightMessage{hascolor=false, ctmax=null, ctmin=null, type=DIMMABLE_LIGHT, state=LightState{reachable=true, on=false, bri=152, alert='none', colormode='null', effect='null', hue=null, sat=null, ct=null, xy=null, transitiontime=null}, e='', r='', t='', id='', manufacturername='NAMRON AS', modelid='4512700', name='Master Bath Roof 70', swversion='2.5.3_r47', ep='', uniqueid='68:0a:e2:ff:fe:3a:34:a5-01'}
2020-12-06 22:09:57.794 [TRACE] [ernal.handler.DeconzBaseThingHandler] - Requesting URL for initial data: http://127.0.0.1:80/api/0950775C33/sensors/3
2020-12-06 22:09:57.796 [TRACE] [ernal.handler.DeconzBaseThingHandler] - Requesting URL for initial data: http://127.0.0.1:80/api/0950775C33/sensors/24
2020-12-06 22:09:57.796 [TRACE] [ernal.handler.DeconzBaseThingHandler] - Requesting URL for initial data: http://127.0.0.1:80/api/0950775C33/sensors/4
2020-12-06 22:09:57.796 [TRACE] [ernal.handler.DeconzBaseThingHandler] - Requesting URL for initial data: http://127.0.0.1:80/api/0950775C33/sensors/7
2020-12-06 22:09:57.796 [TRACE] [ernal.handler.DeconzBaseThingHandler] - Requesting URL for initial data: http://127.0.0.1:80/api/0950775C33/sensors/13
2020-12-06 22:09:57.797 [TRACE] [ernal.handler.DeconzBaseThingHandler] - Requesting URL for initial data: http://127.0.0.1:80/api/0950775C33/sensors/27
```

Lots of sensor data next.
Next part is just before the Java exceptions starts:

```
2020-12-06 22:09:59.444 [TRACE] [ernal.handler.SensorBaseThingHandler] - deconz:switch:homeserver:000b57fffedbaa4a011000 received SensorMessage{type='ZHASwitch', config=SensorConfig{on=true, reachable=false, battery=100, temperature=null, heatsetpoint=null, mode=null, offset=null}, state=SensorState{dark=null, daylight=null, lightlevel=null, lux=null, temperature=null, humidity=null, open=null, fire=null, water=null, alarm=null, tampered=null, vibration=null, carbonmonoxide=null, pressure=null, presence=null, power=null, battery=null, consumption=null, voltage=null, current=null, status=null, buttonevent=3002, gesture=null, valve=null, lastupdated='2020-11-25T07:13:31.366', xy=null}, e='', r='', t='', id='', manufacturername='IKEA of Sweden', modelid='TRADFRI remote control', name='TRÅDFRI remote control Kjøkken', swversion='1.2.223', ep='1', lastseen='null', uniqueid='00:0b:57:ff:fe:db:aa:4a-01-1000'}
2020-12-06 22:09:59.446 [TRACE] [z.internal.handler.LightThingHandler] - deconz:colortemperaturelight:homeserver:000b57fffe103e8101 received LightMessage{hascolor=true, ctmax=454, ctmin=250, type=COLOR_TEMPERATURE_LIGHT, state=LightState{reachable=true, on=false, bri=201, alert='none', colormode='ct', effect='null', hue=null, sat=null, ct=398, xy=null, transitiontime=null}, e='', r='', t='', id='', manufacturername='IKEA of Sweden', modelid='TRADFRI bulb E27 WS opal 980lm', name='Stue pendel', swversion='1.2.217', ep='', uniqueid='00:0b:57:ff:fe:10:3e:81-01'}
2020-12-06 22:09:59.454 [TRACE] [ernal.handler.SensorBaseThingHandler] - deconz:switch:homeserver:000b57fffe1154d4011000 received SensorMessage{type='ZHASwitch', config=SensorConfig{on=true, reachable=false, battery=100, temperature=null, heatsetpoint=null, mode=null, offset=null}, state=SensorState{dark=null, daylight=null, lightlevel=null, lux=null, temperature=null, humidity=null, open=null, fire=null, water=null, alarm=null, tampered=null, vibration=null, carbonmonoxide=null, pressure=null, presence=null, power=null, battery=null, consumption=null, voltage=null, current=null, status=null, buttonevent=1002, gesture=null, valve=null, lastupdated='2020-11-05T14:52:02.174', xy=null}, e='', r='', t='', id='', manufacturername='IKEA of Sweden', modelid='TRADFRI remote control', name='TRÅDFRI remote control Vaskerom', swversion='1.2.223', ep='1', lastseen='null', uniqueid='00:0b:57:ff:fe:11:54:d4-01-1000'}
2020-12-06 22:09:59.456 [TRACE] [z.internal.handler.LightThingHandler] - deconz:extendedcolorlight:homeserver:00178801049a5adb0b received LightMessage{hascolor=true, ctmax=500, ctmin=153, type=EXTENDED_COLOR_LIGHT, state=LightState{reachable=true, on=false, bri=152, alert='none', colormode='ct', effect='none', hue=8418, sat=140, ct=366, xy=[0.4573, 0.41], transitiontime=null}, e='', r='', t='', id='', manufacturername='Philips', modelid='LST002', name='HUE Strip Baderomsinnredning 68', swversion='1.24.0_r20239', ep='', uniqueid='00:17:88:01:04:9a:5a:db-0b'}
2020-12-06 22:09:59.461 [TRACE] [ernal.handler.SensorBaseThingHandler] - deconz:switch:homeserver:000b57fffe11716f011000 received SensorMessage{type='ZHASwitch', config=SensorConfig{on=true, reachable=false, battery=100, temperature=null, heatsetpoint=null, mode=null, offset=null}, state=SensorState{dark=null, daylight=null, lightlevel=null, lux=null, temperature=null, humidity=null, open=null, fire=null, water=null, alarm=null, tampered=null, vibration=null, carbonmonoxide=null, pressure=null, presence=null, power=null, battery=null, consumption=null, voltage=null, current=null, status=null, buttonevent=3002, gesture=null, valve=null, lastupdated='2020-11-25T07:15:14.281', xy=null}, e='', r='', t='', id='', manufacturername='IKEA of Sweden', modelid='TRADFRI remote control', name='TRÅDFRI remote control Hall', swversion='1.2.223', ep='1', lastseen='null', uniqueid='00:0b:57:ff:fe:11:71:6f-01-1000'}
2020-12-06 22:09:59.463 [TRACE] [z.internal.handler.LightThingHandler] - deconz:dimmablelight:homeserver:680ae2fffe3a34a501 received LightMessage{hascolor=false, ctmax=null, ctmin=null, type=DIMMABLE_LIGHT, state=LightState{reachable=true, on=false, bri=152, alert='none', colormode='null', effect='null', hue=null, sat=null, ct=null, xy=null, transitiontime=null}, e='', r='', t='', id='', manufacturername='NAMRON AS', modelid='4512700', name='Master Bath Roof 70', swversion='2.5.3_r47', ep='', uniqueid='68:0a:e2:ff:fe:3a:34:a5-01'}
2020-12-06 22:09:59.468 [TRACE] [ernal.handler.SensorBaseThingHandler] - deconz:switch:homeserver:086bd7fffe5db231011000 received SensorMessage{type='ZHASwitch', config=SensorConfig{on=true, reachable=true, battery=74, temperature=null, heatsetpoint=null, mode=null, offset=null}, state=SensorState{dark=null, daylight=null, lightlevel=null, lux=null, temperature=null, humidity=null, open=null, fire=null, water=null, alarm=null, tampered=null, vibration=null, carbonmonoxide=null, pressure=null, presence=null, power=null, battery=null, consumption=null, voltage=null, current=null, status=null, buttonevent=2002, gesture=null, valve=null, lastupdated='2020-12-06T01:46:20.098', xy=null}, e='', r='', t='', id='', manufacturername='IKEA of Sweden', modelid='TRADFRI open/close remote', name='TRADFRI open/close soverom2', swversion='2.2.008', ep='1', lastseen='2020-12-06T20:51Z', uniqueid='08:6b:d7:ff:fe:5d:b2:31-01-1000'}
2020-12-06 22:09:59.526 [TRACE] [z.internal.handler.LightThingHandler] - deconz:extendedcolorlight:homeserver:00178801040795bf0b received LightMessage{hascolor=true, ctmax=500, ctmin=153, type=EXTENDED_COLOR_LIGHT, state=LightState{reachable=true, on=false, bri=152, alert='none', colormode='ct', effect='none', hue=8418, sat=140, ct=366, xy=[0.4573, 0.41], transitiontime=null}, e='', r='', t='', id='', manufacturername='Philips', modelid='LST002', name='HUE Strip Baderomsinnredning 69', swversion='1.24.0_r20239', ep='', uniqueid='00:17:88:01:04:07:95:bf-0b'}
2020-12-06 22:09:59.530 [TRACE] [ernal.handler.SensorBaseThingHandler] - deconz:switch:homeserver:0017880103a4105e02fc00 received SensorMessage{type='ZHASwitch', config=SensorConfig{on=true, reachable=true, battery=90, temperature=null, heatsetpoint=null, mode=null, offset=null}, state=SensorState{dark=null, daylight=null, lightlevel=null, lux=null, temperature=null, humidity=null, open=null, fire=null, water=null, alarm=null, tampered=null, vibration=null, carbonmonoxide=null, pressure=null, presence=null, power=null, battery=null, consumption=null, voltage=null, current=null, status=null, buttonevent=2002, gesture=null, valve=null, lastupdated='2020-12-06T10:50:24.410', xy=null}, e='', r='', t='', id='', manufacturername='Philips', modelid='RWL021', name='Dimmer Switch', swversion='5.45.1.17846', ep='2', lastseen='2020-12-06T21:05Z', uniqueid='00:17:88:01:03:a4:10:5e-02-fc00'}
2020-12-06 22:09:59.548 [TRACE] [z.internal.handler.LightThingHandler] - deconz:dimmablelight:homeserver:680ae2fffe3a34a501 received LightMessage{hascolor=false, ctmax=null, ctmin=null, type=DIMMABLE_LIGHT, state=LightState{reachable=true, on=false, bri=152, alert='none', colormode='null', effect='null', hue=null, sat=null, ct=null, xy=null, transitiontime=null}, e='', r='', t='', id='', manufacturername='NAMRON AS', modelid='4512700', name='Master Bath Roof 70', swversion='2.5.3_r47', ep='', uniqueid='68:0a:e2:ff:fe:3a:34:a5-01'}
2020-12-06 22:09:59.551 [TRACE] [ernal.handler.SensorBaseThingHandler] - deconz:switch:homeserver:000b57fffe1154d4011000 received SensorMessage{type='ZHASwitch', config=SensorConfig{on=true, reachable=false, battery=100, temperature=null, heatsetpoint=null, mode=null, offset=null}, state=SensorState{dark=null, daylight=null, lightlevel=null, lux=null, temperature=null, humidity=null, open=null, fire=null, water=null, alarm=null, tampered=null, vibration=null, carbonmonoxide=null, pressure=null, presence=null, power=null, battery=null, consumption=null, voltage=null, current=null, status=null, buttonevent=1002, gesture=null, valve=null, lastupdated='2020-11-05T14:52:02.174', xy=null}, e='', r='', t='', id='', manufacturername='IKEA of Sweden', modelid='TRADFRI remote control', name='TRÅDFRI remote control Vaskerom', swversion='1.2.223', ep='1', lastseen='null', uniqueid='00:0b:57:ff:fe:11:54:d4-01-1000'}
2020-12-06 22:09:59.554 [TRACE] [ernal.handler.SensorBaseThingHandler] - deconz:switch:homeserver:000b57fffe0ed885011000 received SensorMessage{type='ZHASwitch', config=SensorConfig{on=true, reachable=true, battery=34, temperature=null, heatsetpoint=null, mode=null, offset=null}, state=SensorState{dark=null, daylight=null, lightlevel=null, lux=null, temperature=null, humidity=null, open=null, fire=null, water=null, alarm=null, tampered=null, vibration=null, carbonmonoxide=null, pressure=null, presence=null, power=null, battery=null, consumption=null, voltage=null, current=null, status=null, buttonevent=5002, gesture=null, valve=null, lastupdated='2020-12-06T11:24:31.785', xy=null}, e='', r='', t='', id='', manufacturername='IKEA of Sweden', modelid='TRADFRI remote control', name='TRADFRI Remote control OMR bed', swversion='1.2.214', ep='1', lastseen='2020-12-06T20:28Z', uniqueid='00:0b:57:ff:fe:0e:d8:85-01-1000'}
2020-12-06 22:09:59.559 [TRACE] [ernal.handler.SensorBaseThingHandler] - deconz:temperaturesensor:homeserver:0017880104b529aa020402 received SensorMessage{type='ZHATemperature', config=SensorConfig{on=true, reachable=true, battery=35, temperature=null, heatsetpoint=null, mode=null, offset=0}, state=SensorState{dark=null, daylight=null, lightlevel=null, lux=null, temperature=2276.0, humidity=null, open=null, fire=null, water=null, alarm=null, tampered=null, vibration=null, carbonmonoxide=null, pressure=null, presence=null, power=null, battery=null, consumption=null, voltage=null, current=null, status=null, buttonevent=null, gesture=null, valve=null, lastupdated='2020-12-06T21:06:46.309', xy=null}, e='', r='', t='', id='', manufacturername='Philips', modelid='SML001', name='Motion Sensor Bad', swversion='6.1.0.18912', ep='2', lastseen='2020-12-06T21:09Z', uniqueid='00:17:88:01:04:b5:29:aa-02-0402'}
2020-12-06 22:09:59.580 [TRACE] [ernal.handler.SensorBaseThingHandler] - deconz:switch:homeserver:086bd7fffe5fa50f011000 received SensorMessage{type='ZHASwitch', config=SensorConfig{on=true, reachable=true, battery=100, temperature=null, heatsetpoint=null, mode=null, offset=null}, state=SensorState{dark=null, daylight=null, lightlevel=null, lux=null, temperature=null, humidity=null, open=null, fire=null, water=null, alarm=null, tampered=null, vibration=null, carbonmonoxide=null, pressure=null, presence=null, power=null, battery=null, consumption=null, voltage=null, current=null, status=null, buttonevent=1002, gesture=null, valve=null, lastupdated='2020-12-06T10:40:29.387', xy=null}, e='', r='', t='', id='', manufacturername='IKEA of Sweden', modelid='TRADFRI open/close remote', name='TRADFRI open/close Soverom1', swversion='2.2.008', ep='1', lastseen='2020-12-06T20:52Z', uniqueid='08:6b:d7:ff:fe:5f:a5:0f-01-1000'}
2020-12-06 22:09:59.590 [TRACE] [ernal.handler.SensorBaseThingHandler] - deconz:daylightsensor:homeserver:00212effff022e8201 received SensorMessage{type='Daylight', config=SensorConfig{on=true, reachable=true, battery=null, temperature=null, heatsetpoint=null, mode=null, offset=null}, state=SensorState{dark=true, daylight=false, lightlevel=null, lux=null, temperature=null, humidity=null, open=null, fire=null, water=null, alarm=null, tampered=null, vibration=null, carbonmonoxide=null, pressure=null, presence=null, power=null, battery=null, consumption=null, voltage=null, current=null, status=230, buttonevent=null, gesture=null, valve=null, lastupdated='2020-12-06T16:58:32.266', xy=null}, e='', r='', t='', id='', manufacturername='Philips', modelid='PHDL00', name='Daylight', swversion='1.0', ep='', lastseen='null', uniqueid='00:21:2e:ff:ff:02:2e:82-01'}
2020-12-06 22:09:59.642 [TRACE] [ernal.handler.SensorBaseThingHandler] - deconz:lightsensor:homeserver:0017880104b529aa020400 received SensorMessage{type='ZHALightLevel', config=SensorConfig{on=true, reachable=true, battery=35, temperature=null, heatsetpoint=null, mode=null, offset=null}, state=SensorState{dark=true, daylight=false, lightlevel=0, lux=0, temperature=null, humidity=null, open=null, fire=null, water=null, alarm=null, tampered=null, vibration=null, carbonmonoxide=null, pressure=null, presence=null, power=null, battery=null, consumption=null, voltage=null, current=null, status=null, buttonevent=null, gesture=null, valve=null, lastupdated='2020-12-06T21:09:36.413', xy=null}, e='', r='', t='', id='', manufacturername='Philips', modelid='SML001', name='Motion Sensor Bad', swversion='6.1.0.18912', ep='2', lastseen='2020-12-06T21:09Z', uniqueid='00:17:88:01:04:b5:29:aa-02-0400'}
2020-12-06 22:09:59.673 [TRACE] [ernal.handler.SensorBaseThingHandler] - deconz:switch:homeserver:00158d00024a4b6803000c received SensorMessage{type='ZHASwitch', config=SensorConfig{on=true, reachable=true, battery=78, temperature=2500.0, heatsetpoint=null, mode=null, offset=null}, state=SensorState{dark=null, daylight=null, lightlevel=null, lux=null, temperature=null, humidity=null, open=null, fire=null, water=null, alarm=null, tampered=null, vibration=null, carbonmonoxide=null, pressure=null, presence=null, power=null, battery=null, consumption=null, voltage=null, current=null, status=null, buttonevent=1058, gesture=7, valve=null, lastupdated='2020-12-05T18:20:54.065', xy=null}, e='', r='', t='', id='', manufacturername='LUMI', modelid='lumi.sensor_cube.aqgl01', name='Mi Magic Cube W1 MasterBath', swversion='20160704', ep='3', lastseen='2020-12-06T20:39Z', uniqueid='00:15:8d:00:02:4a:4b:68-03-000c'}
2020-12-06 22:09:59.700 [TRACE] [ernal.handler.SensorBaseThingHandler] - deconz:switch:homeserver:00158d00010250dc020012 received SensorMessage{type='ZHASwitch', config=SensorConfig{on=true, reachable=true, battery=100, temperature=2100.0, heatsetpoint=null, mode=null, offset=null}, state=SensorState{dark=null, daylight=null, lightlevel=null, lux=null, temperature=null, humidity=null, open=null, fire=null, water=null, alarm=null, tampered=null, vibration=null, carbonmonoxide=null, pressure=null, presence=null, power=null, battery=null, consumption=null, voltage=null, current=null, status=null, buttonevent=7000, gesture=0, valve=null, lastupdated='2020-12-05T11:55:09.413', xy=null}, e='', r='', t='', id='', manufacturername='LUMI', modelid='lumi.sensor_cube', name='Mi Magic Cube P1 Kitchen', swversion='20160407', ep='2', lastseen='2020-12-06T20:39Z', uniqueid='00:15:8d:00:01:02:50:dc-02-0012'}
2020-12-06 22:09:59.807 [DEBUG] [ernal.handler.DeconzBaseThingHandler] - Get new state failed:
java.util.concurrent.CompletionException: java.util.concurrent.TimeoutException: Total timeout 2000 ms elapsed
        at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:331) ~[?:?]
        at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:346) ~[?:?]
        at java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:632) ~[?:?]
        at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506) [?:?]
        at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2088) [?:?]
        at org.openhab.binding.deconz.internal.netutils.AsyncHttpClient$1.onComplete(AsyncHttpClient.java:109) [bundleFile:?]
        at org.eclipse.jetty.client.ResponseNotifier.notifyComplete(ResponseNotifier.java:198) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.client.ResponseNotifier.notifyComplete(ResponseNotifier.java:190) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.client.HttpReceiver.terminateResponse(HttpReceiver.java:444) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.client.HttpReceiver.abort(HttpReceiver.java:527) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.client.HttpChannel.abortResponse(HttpChannel.java:156) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.client.HttpChannel.abort(HttpChannel.java:149) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.client.HttpExchange.abort(HttpExchange.java:257) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.client.HttpConversation.abort(HttpConversation.java:149) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.client.HttpRequest.abort(HttpRequest.java:768) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.client.TimeoutCompleteListener.onTimeoutExpired(TimeoutCompleteListener.java:51) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.client.TimeoutCompleteListener.schedule(TimeoutCompleteListener.java:73) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.client.HttpChannel.send(HttpChannel.java:126) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.client.HttpConnection.send(HttpConnection.java:201) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.client.http.HttpConnectionOverHTTP$Delegate.send(HttpConnectionOverHTTP.java:252) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.client.http.HttpConnectionOverHTTP.send(HttpConnectionOverHTTP.java:121) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.client.http.HttpDestinationOverHTTP.send(HttpDestinationOverHTTP.java:38) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.client.HttpDestination.process(HttpDestination.java:346) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.client.HttpDestination.process(HttpDestination.java:304) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.client.HttpDestination.send(HttpDestination.java:294) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.client.HttpDestination.succeeded(HttpDestination.java:228) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.client.AbstractConnectionPool.proceed(AbstractConnectionPool.java:153) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.client.AbstractConnectionPool$1.succeeded(AbstractConnectionPool.java:131) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.client.AbstractConnectionPool$1.succeeded(AbstractConnectionPool.java:123) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.util.Promise$Wrapper.succeeded(Promise.java:130) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.client.http.HttpConnectionOverHTTP.onOpen(HttpConnectionOverHTTP.java:129) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.io.SelectorManager.connectionOpened(SelectorManager.java:324) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.io.ManagedSelector.createEndPoint(ManagedSelector.java:270) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.io.ManagedSelector.access$1900(ManagedSelector.java:61) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.io.ManagedSelector$CreateEndPoint.run(ManagedSelector.java:923) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:782) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:918) [bundleFile:9.4.20.v20190813]
        at java.lang.Thread.run(Thread.java:834) [?:?]
Caused by: java.util.concurrent.TimeoutException: Total timeout 2000 ms elapsed
        ... 23 more

```

--- a/bundles/org.openhab.binding.deconz/src/main/java/org/openhab/binding/deconz/internal/Util.java
+++ b/bundles/org.openhab.binding.deconz/src/main/java/org/openhab/binding/deconz/internal/Util.java
@@ -25,8 +25,6 @@ import java.util.stream.Stream;
 import org.eclipse.jdt.annotation.NonNullByDefault;
 import org.openhab.core.library.types.DateTimeType;
 import org.openhab.core.library.types.PercentType;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 /**
  * The {@link Util} class defines common utility methods
@@ -35,7 +33,6 @@ import org.slf4j.LoggerFactory;
  */
 @NonNullByDefault
 public class Util {
-    private static final Logger LOGGER = LoggerFactory.getLogger(Util.class);
 
     public static String buildUrl(String host, int port, String... urlParts) {
         StringBuilder url = new StringBuilder();

--- a/bundles/org.openhab.binding.deconz/src/main/java/org/openhab/binding/deconz/internal/discovery/ThingDiscoveryService.java
+++ b/bundles/org.openhab.binding.deconz/src/main/java/org/openhab/binding/deconz/internal/discovery/ThingDiscoveryService.java
@@ -25,7 +25,6 @@ import java.util.stream.Stream;
 import org.eclipse.jdt.annotation.NonNullByDefault;
 import org.eclipse.jdt.annotation.Nullable;
 import org.openhab.binding.deconz.internal.Util;
-import org.openhab.binding.deconz.internal.dto.BridgeFullState;
 import org.openhab.binding.deconz.internal.dto.GroupMessage;
 import org.openhab.binding.deconz.internal.dto.LightMessage;
 import org.openhab.binding.deconz.internal.dto.SensorMessage;
@@ -70,10 +69,19 @@ public class ThingDiscoveryService extends AbstractDiscoveryService implements D
     }
 
     @Override
-    protected void startScan() {
+    public void startScan() {
         final DeconzBridgeHandler handler = this.handler;
         if (handler != null) {
-            handler.requestFullState(false);
+            handler.getBridgeFullState().thenAccept(fullState -> {
+                stopScan();
+                removeOlderResults(getTimestampOfLastScan());
+                fullState.ifPresent(state -> {
+                    state.sensors.forEach(this::addSensor);
+                    state.lights.forEach(this::addLight);
+                    state.groups.forEach(this::addGroup);
+                });
+
+            });
         }
     }
 
@@ -282,7 +290,6 @@ public class ThingDiscoveryService extends AbstractDiscoveryService implements D
     public void setThingHandler(@Nullable ThingHandler handler) {
         if (handler instanceof DeconzBridgeHandler) {
             this.handler = (DeconzBridgeHandler) handler;
-            ((DeconzBridgeHandler) handler).setDiscoveryService(this);
             this.bridgeUID = handler.getThing().getUID();
         }
     }
@@ -300,21 +307,5 @@ public class ThingDiscoveryService extends AbstractDiscoveryService implements D
     @Override
     public void deactivate() {
         super.deactivate();
-    }
-
-    /**
-     * Call this method when a full bridge state request has been performed and either the fullState
-     * are known or a failure happened.
-     *
-     * @param fullState The fullState or null.
-     */
-    public void stateRequestFinished(final @Nullable BridgeFullState fullState) {
-        stopScan();
-        removeOlderResults(getTimestampOfLastScan());
-        if (fullState != null) {
-            fullState.sensors.forEach(this::addSensor);
-            fullState.lights.forEach(this::addLight);
-            fullState.groups.forEach(this::addGroup);
-        }
     }
 }

--- a/bundles/org.openhab.binding.deconz/src/main/java/org/openhab/binding/deconz/internal/dto/BridgeFullState.java
+++ b/bundles/org.openhab.binding.deconz/src/main/java/org/openhab/binding/deconz/internal/dto/BridgeFullState.java
@@ -16,6 +16,8 @@ import java.util.Collections;
 import java.util.Map;
 
 import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.deconz.internal.types.ResourceType;
 
 /**
  * http://dresden-elektronik.github.io/deconz-rest-doc/configuration/
@@ -42,4 +44,17 @@ public class BridgeFullState {
     public Map<String, SensorMessage> sensors = Collections.emptyMap();
     public Map<String, LightMessage> lights = Collections.emptyMap();
     public Map<String, GroupMessage> groups = Collections.emptyMap();
+
+    public @Nullable DeconzBaseMessage getMessage(ResourceType resourceType, String id) {
+        switch (resourceType) {
+            case LIGHTS:
+                return lights.get(id);
+            case SENSORS:
+                return sensors.get(id);
+            case GROUPS:
+                return groups.get(id);
+            default:
+                return null;
+        }
+    }
 }

--- a/bundles/org.openhab.binding.deconz/src/main/java/org/openhab/binding/deconz/internal/handler/DeconzBaseThingHandler.java
+++ b/bundles/org.openhab.binding.deconz/src/main/java/org/openhab/binding/deconz/internal/handler/DeconzBaseThingHandler.java
@@ -14,11 +14,8 @@ package org.openhab.binding.deconz.internal.handler;
 
 import static org.openhab.binding.deconz.internal.Util.buildUrl;
 
-import java.net.SocketTimeoutException;
-import java.util.concurrent.CompletionException;
 import java.util.concurrent.ScheduledFuture;
 import java.util.concurrent.TimeUnit;
-import java.util.concurrent.TimeoutException;
 import java.util.function.Consumer;
 
 import org.eclipse.jdt.annotation.NonNullByDefault;
@@ -46,8 +43,7 @@ import com.google.gson.Gson;
  * @author Jan N. Klug - Refactored to abstract class
  */
 @NonNullByDefault
-public abstract class DeconzBaseThingHandler<T extends DeconzBaseMessage> extends BaseThingHandler
-        implements WebSocketMessageListener {
+public abstract class DeconzBaseThingHandler extends BaseThingHandler implements WebSocketMessageListener {
     private final Logger logger = LoggerFactory.getLogger(DeconzBaseThingHandler.class);
     protected final ResourceType resourceType;
     protected ThingConfig config = new ThingConfig();
@@ -88,6 +84,15 @@ public abstract class DeconzBaseThingHandler<T extends DeconzBaseMessage> extend
         }
     }
 
+    private @Nullable DeconzBridgeHandler getBridgeHandler() {
+        Bridge bridge = getBridge();
+        if (bridge == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+            return null;
+        }
+        return (DeconzBridgeHandler) bridge.getHandler();
+    }
+
     @Override
     public void bridgeStatusChanged(ThingStatusInfo bridgeStatusInfo) {
         if (config.id.isEmpty()) {
@@ -98,12 +103,7 @@ public abstract class DeconzBaseThingHandler<T extends DeconzBaseMessage> extend
         if (bridgeStatusInfo.getStatus() == ThingStatus.ONLINE) {
             // the bridge is ONLINE, we can communicate with the gateway, so we update the connection parameters and
             // register the listener
-            Bridge bridge = getBridge();
-            if (bridge == null) {
-                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
-                return;
-            }
-            DeconzBridgeHandler bridgeHandler = (DeconzBridgeHandler) bridge.getHandler();
+            DeconzBridgeHandler bridgeHandler = getBridgeHandler();
             if (bridgeHandler == null) {
                 updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
                 return;
@@ -130,50 +130,32 @@ public abstract class DeconzBaseThingHandler<T extends DeconzBaseMessage> extend
     }
 
     /**
-     * parse the initial state response message
-     *
-     * @param r AsyncHttpClient.Result with the state response result
-     * @return a message of the correct type
-     */
-    protected abstract @Nullable T parseStateResponse(AsyncHttpClient.Result r);
-
-    /**
      * processes a newly received (initial) state response
      *
      * MUST set the thing status!
      *
      * @param stateResponse
      */
-    protected abstract void processStateResponse(@Nullable T stateResponse);
+    protected abstract void processStateResponse(DeconzBaseMessage stateResponse);
 
     /**
      * Perform a request to the REST API for retrieving the full light state with all data and configuration.
      */
-    protected void requestState(Consumer<@Nullable T> processor) {
-        AsyncHttpClient asyncHttpClient = http;
-        if (asyncHttpClient == null) {
-            return;
+    protected void requestState(Consumer<DeconzBaseMessage> processor) {
+        DeconzBridgeHandler bridgeHandler = getBridgeHandler();
+        if (bridgeHandler != null) {
+            bridgeHandler.getBridgeFullState()
+                    .thenAccept(f -> f.map(s -> s.getMessage(resourceType, config.id)).ifPresentOrElse(message -> {
+                        logger.trace("{} processing {}", thing.getUID(), message);
+                        processor.accept(message);
+                    }, () -> {
+                        if (initializationJob != null) {
+                            stopInitializationJob();
+                            initializationJob = scheduler.schedule(() -> requestState(this::processStateResponse), 10,
+                                    TimeUnit.SECONDS);
+                        }
+                    }));
         }
-
-        String url = buildUrl(bridgeConfig.host, bridgeConfig.httpPort, bridgeConfig.apikey,
-                resourceType.getIdentifier(), config.id);
-        logger.trace("Requesting URL for initial data: {}", url);
-
-        // Get initial data
-        asyncHttpClient.get(url, bridgeConfig.timeout).thenApply(this::parseStateResponse).exceptionally(e -> {
-            if (e instanceof SocketTimeoutException || e instanceof TimeoutException
-                    || e instanceof CompletionException) {
-                logger.debug("Get new state failed: ", e);
-            } else {
-                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
-            }
-
-            stopInitializationJob();
-            initializationJob = scheduler.schedule(() -> requestState(this::processStateResponse), 10,
-                    TimeUnit.SECONDS);
-
-            return null;
-        }).thenAccept(processor);
     }
 
     /**

--- a/bundles/org.openhab.binding.deconz/src/main/java/org/openhab/binding/deconz/internal/handler/DeconzBridgeHandler.java
+++ b/bundles/org.openhab.binding.deconz/src/main/java/org/openhab/binding/deconz/internal/handler/DeconzBridgeHandler.java
@@ -16,10 +16,7 @@ import static org.openhab.binding.deconz.internal.BindingConstants.*;
 import static org.openhab.binding.deconz.internal.Util.buildUrl;
 
 import java.net.SocketTimeoutException;
-import java.util.Collection;
-import java.util.Collections;
-import java.util.Map;
-import java.util.Set;
+import java.util.*;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.CompletionException;
 import java.util.concurrent.ScheduledFuture;
@@ -34,6 +31,7 @@ import org.openhab.binding.deconz.internal.dto.BridgeFullState;
 import org.openhab.binding.deconz.internal.netutils.AsyncHttpClient;
 import org.openhab.binding.deconz.internal.netutils.WebSocketConnection;
 import org.openhab.binding.deconz.internal.netutils.WebSocketConnectionListener;
+import org.openhab.core.cache.ExpiringCacheAsync;
 import org.openhab.core.config.core.Configuration;
 import org.openhab.core.io.net.http.WebSocketFactory;
 import org.openhab.core.thing.Bridge;
@@ -64,7 +62,6 @@ public class DeconzBridgeHandler extends BaseBridgeHandler implements WebSocketC
     public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = Collections.singleton(BRIDGE_TYPE);
 
     private final Logger logger = LoggerFactory.getLogger(DeconzBridgeHandler.class);
-    private @Nullable ThingDiscoveryService thingDiscoveryService;
     private final WebSocketConnection websocket;
     private final AsyncHttpClient http;
     private DeconzBridgeConfig config = new DeconzBridgeConfig();
@@ -74,6 +71,8 @@ public class DeconzBridgeHandler extends BaseBridgeHandler implements WebSocketC
     /** Prevent a dispose/init cycle while this flag is set. Use for property updates */
     private boolean ignoreConfigurationUpdate;
     private boolean websocketReconnect = false;
+
+    private final ExpiringCacheAsync<Optional<BridgeFullState>> fullStateCache = new ExpiringCacheAsync<>(1000);
 
     /** The poll frequency for the API Key verification */
     private static final int POLL_FREQUENCY_SEC = 10;
@@ -130,7 +129,7 @@ public class DeconzBridgeHandler extends BaseBridgeHandler implements WebSocketC
             stopTimer();
             scheduledFuture = scheduler.schedule(() -> requestApiKey(), POLL_FREQUENCY_SEC, TimeUnit.SECONDS);
         } else if (r.getResponseCode() == 200) {
-            ApiKeyMessage[] response = gson.fromJson(r.getBody(), ApiKeyMessage[].class);
+            ApiKeyMessage[] response = Objects.requireNonNull(gson.fromJson(r.getBody(), ApiKeyMessage[].class));
             if (response.length == 0) {
                 throw new IllegalStateException("Authorisation request response is empty");
             }
@@ -139,65 +138,65 @@ public class DeconzBridgeHandler extends BaseBridgeHandler implements WebSocketC
             configuration.put(CONFIG_APIKEY, config.apikey);
             updateConfiguration(configuration);
             updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_PENDING, "Waiting for configuration");
-            requestFullState(true);
+            initializeBridgeState();
         } else {
             throw new IllegalStateException("Unknown status code for authorisation request");
         }
     }
 
     /**
-     * Parses the response message to the REST API for retrieving the full bridge state with all sensors and switches
-     * and configuration.
+     * get the full state of the bridge from the cache
      *
-     * @param r The response
+     * @return a CompletableFuture that returns an Optional of the bridge full state
      */
-    private @Nullable BridgeFullState parseBridgeFullStateResponse(AsyncHttpClient.Result r) {
-        if (r.getResponseCode() == 403) {
-            return null;
-        } else if (r.getResponseCode() == 200) {
-            return gson.fromJson(r.getBody(), BridgeFullState.class);
-        } else {
-            throw new IllegalStateException("Unknown status code for full state request");
+    public CompletableFuture<Optional<BridgeFullState>> getBridgeFullState() {
+        return fullStateCache.getValue(this::refreshFullStateCache);
+    }
+
+    /**
+     * refresh the full bridge state (used for initial processing and state-lookup)
+     *
+     * @return Completable future with an Optional of the BridgeFullState
+     */
+    private CompletableFuture<Optional<BridgeFullState>> refreshFullStateCache() {
+        logger.trace("{} starts refreshing the fullStateCache", thing.getUID());
+        if (config.apikey == null) {
+            return CompletableFuture.completedFuture(Optional.empty());
         }
+        String url = buildUrl(config.getHostWithoutPort(), config.httpPort, config.apikey);
+        return http.get(url, config.timeout).thenApply(r -> {
+            if (r.getResponseCode() == 403) {
+                return Optional.ofNullable((BridgeFullState) null);
+            } else if (r.getResponseCode() == 200) {
+                return Optional.ofNullable(gson.fromJson(r.getBody(), BridgeFullState.class));
+            } else {
+                throw new IllegalStateException("Unknown status code for full state request");
+            }
+        }).handle((v, t) -> {
+            if (t == null) {
+                return v;
+            } else if (t instanceof SocketTimeoutException || t instanceof TimeoutException
+                    || t instanceof CompletionException) {
+                logger.debug("Get full state failed", t);
+            } else if (t != null) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, t.getMessage());
+            }
+            return Optional.empty();
+        });
     }
 
     /**
      * Perform a request to the REST API for retrieving the full bridge state with all sensors and switches
      * and configuration.
      */
-    public void requestFullState(boolean isInitialRequest) {
-        if (config.apikey == null) {
-            return;
-        }
-        String url = buildUrl(config.getHostWithoutPort(), config.httpPort, config.apikey);
-        http.get(url, config.timeout).thenApply(this::parseBridgeFullStateResponse).exceptionally(e -> {
-            if (e instanceof SocketTimeoutException || e instanceof TimeoutException
-                    || e instanceof CompletionException) {
-                logger.debug("Get full state failed", e);
-            } else {
-                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
-            }
-            return null;
-        }).whenComplete((value, error) -> {
-            final ThingDiscoveryService thingDiscoveryService = this.thingDiscoveryService;
-            if (thingDiscoveryService != null) {
-                // Hand over sensors to discovery service
-                thingDiscoveryService.stateRequestFinished(value);
-            }
-        }).thenAccept(fullState -> {
-            if (fullState == null) {
-                if (isInitialRequest) {
-                    scheduledFuture = scheduler.schedule(() -> requestFullState(true), POLL_FREQUENCY_SEC,
-                            TimeUnit.SECONDS);
-                }
-                return;
-            }
-            if (fullState.config.name.isEmpty()) {
+    public void initializeBridgeState() {
+        getBridgeFullState().thenAccept(fullState -> fullState.ifPresentOrElse(state -> {
+            if (state.config.name.isEmpty()) {
                 updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.NONE,
                         "You are connected to a HUE bridge, not a deCONZ software!");
                 return;
             }
-            if (fullState.config.websocketport == 0) {
+            if (state.config.websocketport == 0) {
                 updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.NONE,
                         "deCONZ software too old. No websocket support!");
                 return;
@@ -205,35 +204,37 @@ public class DeconzBridgeHandler extends BaseBridgeHandler implements WebSocketC
 
             // Add some information about the bridge
             Map<String, String> editProperties = editProperties();
-            editProperties.put("apiversion", fullState.config.apiversion);
-            editProperties.put("swversion", fullState.config.swversion);
-            editProperties.put("fwversion", fullState.config.fwversion);
-            editProperties.put("uuid", fullState.config.uuid);
-            editProperties.put("zigbeechannel", String.valueOf(fullState.config.zigbeechannel));
-            editProperties.put("ipaddress", fullState.config.ipaddress);
+            editProperties.put("apiversion", state.config.apiversion);
+            editProperties.put("swversion", state.config.swversion);
+            editProperties.put("fwversion", state.config.fwversion);
+            editProperties.put("uuid", state.config.uuid);
+            editProperties.put("zigbeechannel", String.valueOf(state.config.zigbeechannel));
+            editProperties.put("ipaddress", state.config.ipaddress);
             ignoreConfigurationUpdate = true;
             updateProperties(editProperties);
             ignoreConfigurationUpdate = false;
 
             // Use requested websocket port if no specific port is given
-            websocketPort = config.port == 0 ? fullState.config.websocketport : config.port;
+            websocketPort = config.port == 0 ? state.config.websocketport : config.port;
             websocketReconnect = true;
             startWebsocket();
-        }).exceptionally(e -> {
+        }, () -> {
+            // initial response was empty, re-trying in POLL_FREQUENCY_SEC seconds
+            scheduledFuture = scheduler.schedule(this::initializeBridgeState, POLL_FREQUENCY_SEC, TimeUnit.SECONDS);
+        })).exceptionally(e -> {
             if (e != null) {
                 updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.NONE, e.getMessage());
             } else {
                 updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.NONE);
             }
-            logger.warn("Full state parsing failed", e);
+            logger.warn("Initial full state parsing failed", e);
             return null;
         });
     }
 
     /**
      * Starts the websocket connection.
-     *
-     * {@link #requestFullState} need to be called first to obtain the websocket port.
+     * {@link #initializeBridgeState} need to be called first to obtain the websocket port.
      */
     private void startWebsocket() {
         if (websocket.isConnected() || websocketPort == 0 || websocketReconnect == false) {
@@ -269,7 +270,7 @@ public class DeconzBridgeHandler extends BaseBridgeHandler implements WebSocketC
         if (config.apikey == null) {
             requestApiKey();
         } else {
-            requestFullState(true);
+            initializeBridgeState();
         }
     }
 
@@ -323,14 +324,5 @@ public class DeconzBridgeHandler extends BaseBridgeHandler implements WebSocketC
      */
     public DeconzBridgeConfig getBridgeConfig() {
         return config;
-    }
-
-    /**
-     * Called by the {@link ThingDiscoveryService}. Informs the bridge handler about the service.
-     *
-     * @param thingDiscoveryService The service
-     */
-    public void setDiscoveryService(ThingDiscoveryService thingDiscoveryService) {
-        this.thingDiscoveryService = thingDiscoveryService;
     }
 }

--- a/bundles/org.openhab.binding.deconz/src/main/java/org/openhab/binding/deconz/internal/handler/GroupThingHandler.java
+++ b/bundles/org.openhab.binding.deconz/src/main/java/org/openhab/binding/deconz/internal/handler/GroupThingHandler.java
@@ -17,13 +17,11 @@ import static org.openhab.binding.deconz.internal.BindingConstants.*;
 import java.util.Set;
 
 import org.eclipse.jdt.annotation.NonNullByDefault;
-import org.eclipse.jdt.annotation.Nullable;
 import org.openhab.binding.deconz.internal.Util;
 import org.openhab.binding.deconz.internal.dto.DeconzBaseMessage;
 import org.openhab.binding.deconz.internal.dto.GroupAction;
 import org.openhab.binding.deconz.internal.dto.GroupMessage;
 import org.openhab.binding.deconz.internal.dto.GroupState;
-import org.openhab.binding.deconz.internal.netutils.AsyncHttpClient;
 import org.openhab.binding.deconz.internal.types.ResourceType;
 import org.openhab.core.library.types.DecimalType;
 import org.openhab.core.library.types.HSBType;
@@ -55,7 +53,7 @@ import com.google.gson.Gson;
  * @author Jan N. Klug - Initial contribution
  */
 @NonNullByDefault
-public class GroupThingHandler extends DeconzBaseThingHandler<GroupMessage> {
+public class GroupThingHandler extends DeconzBaseThingHandler {
     public static final Set<ThingTypeUID> SUPPORTED_THING_TYPE_UIDS = Set.of(THING_TYPE_LIGHTGROUP);
     private final Logger logger = LoggerFactory.getLogger(GroupThingHandler.class);
 
@@ -128,21 +126,7 @@ public class GroupThingHandler extends DeconzBaseThingHandler<GroupMessage> {
     }
 
     @Override
-    protected @Nullable GroupMessage parseStateResponse(AsyncHttpClient.Result r) {
-        if (r.getResponseCode() == 403) {
-            return null;
-        } else if (r.getResponseCode() == 200) {
-            return gson.fromJson(r.getBody(), GroupMessage.class);
-        } else {
-            throw new IllegalStateException("Unknown status code " + r.getResponseCode() + " for full state request");
-        }
-    }
-
-    @Override
-    protected void processStateResponse(@Nullable GroupMessage stateResponse) {
-        if (stateResponse == null) {
-            return;
-        }
+    protected void processStateResponse(DeconzBaseMessage stateResponse) {
         messageReceived(config.id, stateResponse);
     }
 

--- a/bundles/org.openhab.binding.deconz/src/main/java/org/openhab/binding/deconz/internal/handler/LightThingHandler.java
+++ b/bundles/org.openhab.binding.deconz/src/main/java/org/openhab/binding/deconz/internal/handler/LightThingHandler.java
@@ -27,7 +27,6 @@ import org.openhab.binding.deconz.internal.Util;
 import org.openhab.binding.deconz.internal.dto.DeconzBaseMessage;
 import org.openhab.binding.deconz.internal.dto.LightMessage;
 import org.openhab.binding.deconz.internal.dto.LightState;
-import org.openhab.binding.deconz.internal.netutils.AsyncHttpClient;
 import org.openhab.binding.deconz.internal.types.ResourceType;
 import org.openhab.core.library.types.*;
 import org.openhab.core.thing.ChannelUID;
@@ -58,7 +57,7 @@ import com.google.gson.Gson;
  * @author Jan N. Klug - Initial contribution
  */
 @NonNullByDefault
-public class LightThingHandler extends DeconzBaseThingHandler<LightMessage> {
+public class LightThingHandler extends DeconzBaseThingHandler {
     public static final Set<ThingTypeUID> SUPPORTED_THING_TYPE_UIDS = Set.of(THING_TYPE_COLOR_TEMPERATURE_LIGHT,
             THING_TYPE_DIMMABLE_LIGHT, THING_TYPE_COLOR_LIGHT, THING_TYPE_EXTENDED_COLOR_LIGHT, THING_TYPE_ONOFF_LIGHT,
             THING_TYPE_WINDOW_COVERING, THING_TYPE_WARNING_DEVICE, THING_TYPE_DOORLOCK);
@@ -263,41 +262,31 @@ public class LightThingHandler extends DeconzBaseThingHandler<LightMessage> {
     }
 
     @Override
-    protected @Nullable LightMessage parseStateResponse(AsyncHttpClient.Result r) {
-        if (r.getResponseCode() == 403) {
-            return null;
-        } else if (r.getResponseCode() == 200) {
-            LightMessage lightMessage = Objects.requireNonNull(gson.fromJson(r.getBody(), LightMessage.class));
-            if (needsPropertyUpdate) {
-                // if we did not receive an ctmin/ctmax, then we probably don't need it
-                needsPropertyUpdate = false;
-
-                Integer ctmax = lightMessage.ctmax;
-                Integer ctmin = lightMessage.ctmin;
-                if (ctmin != null && ctmax != null) {
-                    Map<String, String> properties = new HashMap<>(thing.getProperties());
-                    properties.put(PROPERTY_CT_MAX,
-                            Integer.toString(Util.constrainToRange(ctmax, ZCL_CT_MIN, ZCL_CT_MAX)));
-                    properties.put(PROPERTY_CT_MIN,
-                            Integer.toString(Util.constrainToRange(ctmin, ZCL_CT_MIN, ZCL_CT_MAX)));
-                    updateProperties(properties);
-                }
-            }
-            return lightMessage;
-        } else {
-            throw new IllegalStateException("Unknown status code " + r.getResponseCode() + " for full state request");
-        }
-    }
-
-    @Override
-    protected void processStateResponse(@Nullable LightMessage stateResponse) {
-        if (stateResponse == null) {
+    protected void processStateResponse(DeconzBaseMessage stateResponse) {
+        if (!(stateResponse instanceof LightMessage)) {
             return;
         }
-        if (stateResponse.state.effect != null) {
-            checkAndUpdateEffectChannels(stateResponse);
+
+        LightMessage lightMessage = (LightMessage) stateResponse;
+        if (needsPropertyUpdate) {
+            // if we did not receive an ctmin/ctmax, then we probably don't need it
+            needsPropertyUpdate = false;
+
+            Integer ctmax = lightMessage.ctmax;
+            Integer ctmin = lightMessage.ctmin;
+            if (ctmin != null && ctmax != null) {
+                Map<String, String> properties = new HashMap<>(thing.getProperties());
+                properties.put(PROPERTY_CT_MAX, Integer.toString(Util.constrainToRange(ctmax, ZCL_CT_MIN, ZCL_CT_MAX)));
+                properties.put(PROPERTY_CT_MIN, Integer.toString(Util.constrainToRange(ctmin, ZCL_CT_MIN, ZCL_CT_MAX)));
+                updateProperties(properties);
+            }
         }
-        messageReceived(config.id, stateResponse);
+
+        if (lightMessage.state.effect != null) {
+            checkAndUpdateEffectChannels(lightMessage);
+        }
+
+        messageReceived(config.id, lightMessage);
     }
 
     private enum EffectLightModel {

--- a/bundles/org.openhab.binding.deconz/src/main/java/org/openhab/binding/deconz/internal/handler/SensorBaseThingHandler.java
+++ b/bundles/org.openhab.binding.deconz/src/main/java/org/openhab/binding/deconz/internal/handler/SensorBaseThingHandler.java
@@ -28,7 +28,6 @@ import org.openhab.binding.deconz.internal.dto.DeconzBaseMessage;
 import org.openhab.binding.deconz.internal.dto.SensorConfig;
 import org.openhab.binding.deconz.internal.dto.SensorMessage;
 import org.openhab.binding.deconz.internal.dto.SensorState;
-import org.openhab.binding.deconz.internal.netutils.AsyncHttpClient;
 import org.openhab.binding.deconz.internal.types.ResourceType;
 import org.openhab.core.library.types.DecimalType;
 import org.openhab.core.library.types.OnOffType;
@@ -63,7 +62,7 @@ import com.google.gson.Gson;
  * @author Lukas Agethen - Refactored to provide better extensibility
  */
 @NonNullByDefault
-public abstract class SensorBaseThingHandler extends DeconzBaseThingHandler<SensorMessage> {
+public abstract class SensorBaseThingHandler extends DeconzBaseThingHandler {
     private final Logger logger = LoggerFactory.getLogger(SensorBaseThingHandler.class);
     /**
      * The sensor state. Contains all possible fields for all supported sensors and switches
@@ -106,25 +105,15 @@ public abstract class SensorBaseThingHandler extends DeconzBaseThingHandler<Sens
     }
 
     @Override
-    protected @Nullable SensorMessage parseStateResponse(AsyncHttpClient.Result r) {
-        if (r.getResponseCode() == 403) {
-            return null;
-        } else if (r.getResponseCode() == 200) {
-            return gson.fromJson(r.getBody(), SensorMessage.class);
-        } else {
-            throw new IllegalStateException("Unknown status code " + r.getResponseCode() + " for full state request");
-        }
-    }
-
-    @Override
-    protected void processStateResponse(@Nullable SensorMessage stateResponse) {
-        logger.trace("{} received {}", thing.getUID(), stateResponse);
-        if (stateResponse == null) {
+    protected void processStateResponse(DeconzBaseMessage stateResponse) {
+        if (!(stateResponse instanceof SensorMessage)) {
             return;
         }
-        SensorConfig newSensorConfig = stateResponse.config;
+
+        SensorMessage sensorMessage = (SensorMessage) stateResponse;
+        SensorConfig newSensorConfig = sensorMessage.config;
         sensorConfig = newSensorConfig != null ? newSensorConfig : new SensorConfig();
-        SensorState newSensorState = stateResponse.state;
+        SensorState newSensorState = sensorMessage.state;
         sensorState = newSensorState != null ? newSensorState : new SensorState();
 
         // Add some information about the sensor
@@ -139,9 +128,9 @@ public abstract class SensorBaseThingHandler extends DeconzBaseThingHandler<Sens
         }
 
         Map<String, String> editProperties = editProperties();
-        editProperties.put(Thing.PROPERTY_FIRMWARE_VERSION, stateResponse.swversion);
-        editProperties.put(Thing.PROPERTY_MODEL_ID, stateResponse.modelid);
-        editProperties.put(UNIQUE_ID, stateResponse.uniqueid);
+        editProperties.put(Thing.PROPERTY_FIRMWARE_VERSION, sensorMessage.swversion);
+        editProperties.put(Thing.PROPERTY_MODEL_ID, sensorMessage.modelid);
+        editProperties.put(UNIQUE_ID, sensorMessage.uniqueid);
         ignoreConfigurationUpdate = true;
         updateProperties(editProperties);
 
@@ -166,7 +155,7 @@ public abstract class SensorBaseThingHandler extends DeconzBaseThingHandler<Sens
         // So to monitor a sensor is still alive, the "last seen" is necessary.
         // Because "last seen" is never updated by the WebSocket API - if this is supported, then we have to
         // manually poll it after the defined time
-        String lastSeen = stateResponse.lastseen;
+        String lastSeen = sensorMessage.lastseen;
         if (lastSeen != null && config.lastSeenPolling > 0) {
             createChannel(CHANNEL_LAST_SEEN, ChannelKind.STATE);
             updateState(CHANNEL_LAST_SEEN, Util.convertTimestampToDateTime(lastSeen));
@@ -179,10 +168,7 @@ public abstract class SensorBaseThingHandler extends DeconzBaseThingHandler<Sens
         updateStatus(ThingStatus.ONLINE);
     }
 
-    private void processLastSeen(@Nullable SensorMessage stateResponse) {
-        if (stateResponse == null) {
-            return;
-        }
+    private void processLastSeen(DeconzBaseMessage stateResponse) {
         String lastSeen = stateResponse.lastseen;
         if (lastSeen != null) {
             updateState(CHANNEL_LAST_SEEN, Util.convertTimestampToDateTime(lastSeen));

--- a/bundles/org.openhab.binding.deconz/src/main/java/org/openhab/binding/deconz/internal/netutils/WebSocketConnection.java
+++ b/bundles/org.openhab.binding.deconz/src/main/java/org/openhab/binding/deconz/internal/netutils/WebSocketConnection.java
@@ -26,9 +26,6 @@ import org.eclipse.jetty.websocket.api.annotations.OnWebSocketMessage;
 import org.eclipse.jetty.websocket.api.annotations.WebSocket;
 import org.eclipse.jetty.websocket.client.WebSocketClient;
 import org.openhab.binding.deconz.internal.dto.DeconzBaseMessage;
-import org.openhab.binding.deconz.internal.dto.GroupMessage;
-import org.openhab.binding.deconz.internal.dto.LightMessage;
-import org.openhab.binding.deconz.internal.dto.SensorMessage;
 import org.openhab.binding.deconz.internal.types.ResourceType;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
@@ -45,10 +42,6 @@ import com.google.gson.Gson;
 @WebSocket
 @NonNullByDefault
 public class WebSocketConnection {
-    private static final Map<ResourceType, Class<? extends DeconzBaseMessage>> EXPECTED_MESSAGE_TYPES = Map.of(
-            ResourceType.GROUPS, GroupMessage.class, ResourceType.LIGHTS, LightMessage.class, ResourceType.SENSORS,
-            SensorMessage.class);
-
     private final Logger logger = LoggerFactory.getLogger(WebSocketConnection.class);
 
     private final WebSocketClient client;
@@ -130,7 +123,7 @@ public class WebSocketConnection {
             return;
         }
 
-        Class<? extends DeconzBaseMessage> expectedMessageType = EXPECTED_MESSAGE_TYPES.get(changedMessage.r);
+        Class<? extends DeconzBaseMessage> expectedMessageType = changedMessage.r.getExpectedMessageType();
         if (expectedMessageType == null) {
             logger.warn("BUG! Could not get expected message type for resource type {}. Please report this incident.",
                     changedMessage.r);

--- a/bundles/org.openhab.binding.deconz/src/main/java/org/openhab/binding/deconz/internal/types/ResourceType.java
+++ b/bundles/org.openhab.binding.deconz/src/main/java/org/openhab/binding/deconz/internal/types/ResourceType.java
@@ -17,6 +17,11 @@ import java.util.Map;
 import java.util.stream.Collectors;
 
 import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.deconz.internal.dto.DeconzBaseMessage;
+import org.openhab.binding.deconz.internal.dto.GroupMessage;
+import org.openhab.binding.deconz.internal.dto.LightMessage;
+import org.openhab.binding.deconz.internal.dto.SensorMessage;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
@@ -27,10 +32,10 @@ import org.slf4j.LoggerFactory;
  */
 @NonNullByDefault
 public enum ResourceType {
-    GROUPS("groups", "action"),
-    LIGHTS("lights", "state"),
-    SENSORS("sensors", "config"),
-    UNKNOWN("", "");
+    GROUPS("groups", "action", GroupMessage.class),
+    LIGHTS("lights", "state", LightMessage.class),
+    SENSORS("sensors", "config", SensorMessage.class),
+    UNKNOWN("", "", null);
 
     private static final Map<String, ResourceType> MAPPING = Arrays.stream(ResourceType.values())
             .collect(Collectors.toMap(v -> v.identifier, v -> v));
@@ -38,10 +43,13 @@ public enum ResourceType {
 
     private String identifier;
     private String commandUrl;
+    private @Nullable Class<? extends DeconzBaseMessage> expectedMessageType;
 
-    ResourceType(String identifier, String commandUrl) {
+    ResourceType(String identifier, String commandUrl,
+            @Nullable Class<? extends DeconzBaseMessage> expectedMessageType) {
         this.identifier = identifier;
         this.commandUrl = commandUrl;
+        this.expectedMessageType = expectedMessageType;
     }
 
     /**
@@ -60,6 +68,15 @@ public enum ResourceType {
      */
     public String getCommandUrl() {
         return commandUrl;
+    }
+
+    /**
+     * get the expected message type for this resource type
+     *
+     * @return
+     */
+    public @Nullable Class<? extends DeconzBaseMessage> getExpectedMessageType() {
+        return expectedMessageType;
     }
 
     /**

--- a/bundles/org.openhab.binding.deconz/src/test/java/org/openhab/binding/deconz/DeconzTest.java
+++ b/bundles/org.openhab.binding.deconz/src/test/java/org/openhab/binding/deconz/DeconzTest.java
@@ -20,6 +20,8 @@ import java.io.IOException;
 import java.nio.charset.StandardCharsets;
 import java.time.ZoneId;
 import java.time.ZonedDateTime;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
 
 import org.eclipse.jdt.annotation.NonNullByDefault;
 import org.junit.jupiter.api.BeforeEach;
@@ -78,11 +80,12 @@ public class DeconzTest {
         assertEquals(6, bridgeFullState.lights.size());
         assertEquals(9, bridgeFullState.sensors.size());
 
+        Mockito.doAnswer(answer -> CompletableFuture.completedFuture(Optional.of(bridgeFullState))).when(bridgeHandler)
+                .getBridgeFullState();
         ThingDiscoveryService discoveryService = new ThingDiscoveryService();
         discoveryService.setThingHandler(bridgeHandler);
         discoveryService.addDiscoveryListener(discoveryListener);
-
-        discoveryService.stateRequestFinished(bridgeFullState);
+        discoveryService.startScan();
         Mockito.verify(discoveryListener, times(20)).thingDiscovered(any(), any());
     }
 

